client/components/language-picker/modal.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { ReactNode, useState } from 'react';
+import { Dialog } from '@automattic/components';
+import { useI18n } from '@automattic/react-i18n';
+import LanguagePicker, { Language, createLanguageGroups } from '@automattic/language-picker';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+
+/**
+ * Style dependencies
+ */
+import './modal.scss';
+
+type Props = {
+	languages: Language[];
+	/**
+	 * Fires when the modal is dismissed.
+	 */
+	onClose: () => void;
+	/**
+	 * Fires when the modal is dismissed with a language selection.
+	 */
+	onSelectLanguage: ( l: Language ) => void;
+	selectedLanguageSlug?: string;
+};
+
+const LanguagePickerModal = ( {
+	languages,
+	onClose,
+	onSelectLanguage,
+	selectedLanguageSlug,
+}: Props ): ReactNode => {
+	const { __ } = useI18n();
+	const [ selectedLanguage, setSelectedLanguage ] = useState< Language | undefined >(
+		languages.find( ( l ) => l.langSlug === selectedLanguageSlug )
+	);
+
+	const buttons = [
+		<Button isLink isLarge onClick={ onClose }>
+			{ __( 'Cancel' ) }
+		</Button>,
+		<Button
+			isSecondary
+			isLarge
+			onClick={ () => {
+				onClose();
+				if ( selectedLanguage ) {
+					onSelectLanguage( selectedLanguage );
+				}
+			} }
+		>
+			{ __( 'Apply Changes' ) }
+		</Button>,
+	];
+
+	return (
+		<Dialog
+			isVisible
+			onClose={ onClose }
```

Comment:
We're also removing an existing [tracking event](https://github.com/Automattic/wp-calypso/pull/46328/files#diff-85febdb2d08b1d64ca197b57b3012332ac71b2a8d72d91f43be2257e136f7479L418).

Reply:
A good catch, we'll have to reimplement it using the provided hooks.